src/main/ngapp/accio/src/app/editor/mask.directive.ts

```diff
@@ -173,46 +172,17 @@ export class MaskDirective {
         && this.scribbleFill) {
 
       this.scribbleFill = false;
-      const maskPixels = this.magicWandService.scribbleFloodfill(
-        this.originalImageData,
-        this.coord[0],
-        this.coord[1],
-        this.tolerance,
-        this.paintPixels
-      );
+      this.scribbleFillEvent.emit(new Coordinate(this.coord[0], this.coord[1]));
+    }
 
-      this.newMaskEvent.emit(
-        new Mask.MaskAction(
-          this.tool == MaskTool.MAGIC_WAND_ADD
-            ? Mask.Action.ADD
-            : Mask.Action.SUBTRACT,
-          Mask.Tool.SCRIBBLE,
-          maskPixels
-        )
-      );
-    } else if (
+    else if (
       (this.tool == MaskTool.MAGIC_WAND_ADD ||
         this.tool == MaskTool.MAGIC_WAND_SUB) &&
       !this.scribbleFill
     ) {
-      // Returns an array indices of each pixel in the mask.
-      const maskPixels = this.magicWandService.floodfill(
-        this.originalImageData,
-        this.coord[0],
-        this.coord[1],
-        this.tolerance
-      );
-
-      this.newMaskEvent.emit(
-        new Mask.MaskAction(
-          this.tool == MaskTool.MAGIC_WAND_ADD
-            ? Mask.Action.ADD
-            : Mask.Action.SUBTRACT,
-          Mask.Tool.MAGIC_WAND,
-          maskPixels
-        )
-      );
+      this.floodfillEvent.emit(new Coordinate(this.coord[0], this.coord[1]));
     }
```

Comment:
#67 Same here

Reply:
Right, probably something we will have to do once yours is branched into master. I knew it would be an issue I was just hoping it wouldn't be too bad. I think we could figure it out depending on how much time there is tomorrow.